test/geometry_tests.cpp

```diff
@@ -45,6 +45,39 @@ bool allXOrYCoorSame(int x_or_y, std::vector<std::vector<int>> pixels){
 
 }
 
+bool allXOrYCoorSame(int x_or_y, std::vector<Cell> v){
+	int shouldBeEqualTo;
+	if (x_or_y == 0) { // Check X coordinates
+		x_or_y = 0;
+		shouldBeEqualTo = v[0].x;
+	} else { // Check Y coordinatesd
+		x_or_y = 1;
+		shouldBeEqualTo = v[0].y;
+	}
+
+	float sum = 0;
+
+	if (x_or_y){
+		for (const auto &pixel:v) {
+			sum += pixel.y;
+		}
+	} else {
+		for (const auto &pixel:v) {
+			sum += pixel.x;
+		}
+	}
+
+
+	float avg = sum/(float)v.size();
+
+	if (avg == shouldBeEqualTo) {
+		return true;
+	} else {
+		return false;
+	}
+
+}
+
 bool checkFirstOrLastPoint(int firstOrLastPoint, int x_shouldbe, int y_shouldbe, std::vector<std::vector<int>> v) {
```

Comment:
I clearly missing the existence of this function last PR. What's the point here?

Reply:
Removed